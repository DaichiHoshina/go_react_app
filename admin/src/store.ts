import { combineReducers } from 'redux';
import logger from 'redux-logger';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { userSlice, initialState as userInitialState } from './modules/User';

const rootReducer = combineReducers({
  userState: userSlice.reducer,
});

const preloadedState = () => {
  return { userState: userInitialState };
};

const store = () => {
  const middlewareList = [...getDefaultMiddleware(), logger];

  return configureStore({
    reducer: rootReducer,
    middleware: middlewareList,
    devTools: process.env.NODE_ENV !== 'production',
    preloadedState: preloadedState(),
  });
};

export default store;
