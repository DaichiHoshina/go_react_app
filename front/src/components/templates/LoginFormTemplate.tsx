import { FC } from "react";
import LoginForm from "../organisms/login/LoginForm";

const LoginFormTemplate: FC = () => {
  return (
    <>
      <div className="flex justify-center p-24">
        <LoginForm isSubmit={false} />
      </div>
    </>
  );
};
export default LoginFormTemplate;
