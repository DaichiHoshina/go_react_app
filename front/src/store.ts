import { combineReducers } from "redux";
import logger from "redux-logger";
import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import { userSlice, initialState as userInitialState } from "./modules/User";
import {
  presentationSlice,
  initialState as presentationInitialState,
} from "./modules/Presentation";

const rootReducer = combineReducers({
  userState: userSlice.reducer,
  presentationState: presentationSlice.reducer,
});

const preloadedState = () => {
  return {
    userState: userInitialState,
    presentationState: presentationInitialState,
  };
};

const store = () => {
  const middlewareList = [...getDefaultMiddleware(), logger];

  return configureStore({
    reducer: rootReducer,
    middleware: middlewareList,
    devTools: process.env.NODE_ENV !== "production",
    preloadedState: preloadedState(),
  });
};

export default store;
