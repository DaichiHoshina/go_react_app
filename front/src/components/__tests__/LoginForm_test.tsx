import "@testing-library/jest-dom/extend-expect";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../../store";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import LoginFormTemplate from "../templates/LoginFormTemplate";

it("<LoginTemplate>の新規作成画面スナップショット", () => {
  const tree = renderer
    .create(
      <Provider store={store()}>
        <LoginFormTemplate />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("ユーザー登録のレンダリングとボタン押下時", () => {
  render(
    <Provider store={store()}>
      <LoginFormTemplate />
    </Provider>
  );

  const emailInput = screen.getByText("email");
  userEvent.type(emailInput, "testtest@test.jp");

  const passwordInput = screen.getByText("password");
  userEvent.type(passwordInput, "test1234");

  const submit = screen.getByTestId("form");
  expect(submit).toHaveClass("Mui-disabled");
});
