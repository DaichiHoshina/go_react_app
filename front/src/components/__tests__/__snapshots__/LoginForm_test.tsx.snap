// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginTemplate>の新規作成画面スナップショット 1`] = `
<div
  className="flex justify-center p-24"
>
  <div
    className="MuiPaper-root MuiCard-root flex-item flex flex-col px-8 py-12 w-1/2 max-w-5xl MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "backgroundColor": "#242323",
        "color": "#ffffff",
      }
    }
  >
    <h1
      className="pb-8 text-center text-xl"
    >
      Sign In
    </h1>
    <form
      action=""
      onSubmit={[Function]}
    >
      <ul
        className="pb-10 space-y-3"
      >
        <li
          className=""
        >
          <p
            className="w-48"
          >
            email
            <span
              className="mx-3"
            >
              :
            </span>
          </p>
          <div
            className="bg-gray-700"
          >
            <div
              className="w-full"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="outlined-basic"
                    name="email"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder=""
                    required={false}
                    rows={1}
                    style={
                      Object {
                        "padding": 8,
                        "paddingLeft": 10,
                        "paddingRight": 10,
                      }
                    }
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-2"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          className=""
        >
          <p
            className="w-48"
          >
            password
            <span
              className="mx-3"
            >
              :
            </span>
          </p>
          <div
            className="bg-gray-700"
          >
            <div
              className="w-full"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    disabled={false}
                    id="outlined-basic"
                    name="password"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder=""
                    required={false}
                    rows={1}
                    style={
                      Object {
                        "padding": 8,
                        "paddingLeft": 10,
                        "paddingRight": 10,
                      }
                    }
                    type="password"
                    value=""
                  />
                  <div
                    className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      className="MuiButtonBase-root MuiIconButton-root"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-2"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained border-none ring-transparent MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
        data-testid="form"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
      </button>
      <div
        className="mt-5"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained border-none ring-transparent MuiButton-containedSecondary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Login Test User!!
          </span>
        </button>
      </div>
      <div>
        ※ポートフォリオ閲覧用です
      </div>
    </form>
  </div>
</div>
`;
